client/starwhale/core/dataset/store.py

```diff
@@ -218,55 +226,88 @@ def __init__(
 
     __repr__ = __str__
 
     @classmethod
-    def from_uri(cls, uri: str, auth_name: str = "") -> S3Connection:
+    def from_uri(cls, uri: str) -> S3Connection:
         """make S3 Connection by uri
 
         uri:
             - s3://username:password@127.0.0.1:8000/bucket/key
             - s3://127.0.0.1:8000/bucket/key
         """
-        from .type import S3LinkAuth
-
-        uri = uri.strip()
-        if not uri or not uri.startswith(("s3://", "minio://")):
+        r = urlparse(uri.strip())
+        if not r.scheme or r.scheme not in cls.supported_schemes:
             raise NoSupportError(
-                f"s3 connection only support s3:// prefix, the actual uri is {uri}"
+                f"s3 connection only support {cls.supported_schemes} prefix, the actual uri is {uri}"
             )
-
-        r = urlparse(uri)
-
-        link_auth = S3LinkAuth.from_env(auth_name)
-        access = r.username or link_auth.access_key
-        secret = r.password or link_auth.secret
-        region = link_auth.region
-        endpoint = r.netloc.split("@")[-1] or link_auth.endpoint
-
+        endpoint_loc = r.netloc.split("@")[-1]
         parts = r.path.lstrip("/").split("/", 1)
         if len(parts) != 2 or parts[0] == "" or parts[1] == "":
             raise FieldTypeOrValueError(
                 f"{uri} is not a valid s3 uri for bucket and key"
             )
         bucket = parts[0]
+        cls.build_init_connections()
+        for connection in S3Connection.connections_config:
+            if connection.fits(bucket, endpoint_loc):
+                return connection
+        if r.username and r.password:
+            return cls(
+                endpoint=endpoint_loc,
+                access_key=r.username,
+                secret_key=r.password,
+                region=_DEFAULT_S3_REGION,
+                bucket=bucket,
+            )
+        raise NoSupportError(f"no matching s3 config in {get_swcli_config_path()}")
 
-        if not endpoint:
-            raise FieldTypeOrValueError("endpoint is empty")
-
-        if not access or not secret:
-            raise FieldTypeOrValueError("no access_key or secret_key")
-
-        return cls(
-            endpoint=endpoint,
-            access_key=access,
-            secret_key=secret,
-            region=region or _DEFAULT_S3_REGION,
-            bucket=bucket,
-        )
+    @classmethod
+    def build_init_connections(cls) -> None:
+        with S3Connection.init_config_lock:
+            if S3Connection.connections_config:
+                return
+            sw_config = SWCliConfigMixed()
+            link_auths = sw_config.link_auths
+            if not link_auths:
+                return
+            for la in link_auths:
+                if not la.get("type") or type(la.get("type")) != str:
+                    continue
+                if not la.get("type") in cls.supported_schemes:
+                    continue
```

Comment:
`if not la.get("type") in cls.supported_schemes:`
-->
`if la.get("type") not in cls.supported_schemes:`
?

Reply:
> `if not la.get("type") in cls.supported_schemes:` --> `if la.get("type") not in cls.supported_schemes:` ?

done